### PR TITLE
various bugfixes and tests

### PR DESCRIPTION
**Pull request by maraoz**
- fixes https://github.com/bitpay/bitcore/issues/1018
- fixes https://github.com/bitpay/bitcore/issues/711
- fixes ipv4/ipv6 parsing
- adds inv/getdata integration tests
- fix `version` and `addr` messages parsing/serialization
- added preconditions to check parsing is correct
- convert many Buffer members to more friendly types (objects, numbers, etc)
- increase test coverage

### CODE REVIEW

**Comment by maraoz** [comment]
I'll fix coverage decrease first

File: integration/bitcoind.js [hunk 1]
```diff
@@ -143,39 +143,39 @@ describe('Integration with ' + network.name + ' bitcoind', function() {
...
   var from = [blockHash[network.name]];
   var stop = stopBlock[network.name];
-  it('can get headers', function(cb) {
+  it('gets headers', function(cb) {
     connect(function(peer) {
       peer.once('headers', function(message) {
         (message instanceof Messages.Headers).should.equal(true);
-        message.headers.length.should.equal(2);
+        message.headers.length.should.equal(3);
+        //console.log(message.serialize(network).toString('hex'));
```
Comment (eordano) [inline integration/bitcoind.js]: please remove?

**Comment by eordano** [comment]
LGTM

**Comment by yemel** [comment]
LGTM

**Comment by maraoz** [comment]
updated

**Pull request merged**